homeassistant/components/simplisafe/__init__.py

```diff
@@ -200,3 +184,33 @@ def _async_save_refresh_token(hass, config_entry, token):
     remove_listener()
 
     return True
+
+
+class SimpliSafe:
+    """Define a SimpliSafe API object."""
+
+    def __init__(self, hass, config_entry, systems):
+        """Initialize."""
+        self._config_entry = config_entry
+        self._hass = hass
+        self.systems = systems
+
+    async def async_update(self):
+        """Get updated data from SimpliSafe."""
+        tasks = [system.update() for system in self.systems.values()]
+
+        results = await asyncio.gather(*tasks, return_exceptions=True)
+        for system, result in zip(self.systems.values(), results):
```

Comment:
Is systems an ordered dict? I'm thinking there could be a mismatch here unless it's an ordered dict. We could cache the systems values in a variable first above so we don't access the dict twice.